numpy/random/mtrand.pyx

```diff
@@ -3769,7 +3831,11 @@ cdef class RandomState:
                     raise ValueError(
                         "covariance is not positive-semidefinite.")
 
-        x = np.dot(x, np.sqrt(s)[:, None] * v)
+        if mode == 'fast':
+            x = np.dot(x, l)
+        else:
+            s = np.abs(s) # to combat negative eigenvalues
```

Comment:
Is this well motivated? If the matrix isn't psd and you're not checking that fact should numpy be trying to silently correct the eigenvalues? IDK, but possibly deserves discussion...

Reply:
This was done following suggestions made in the issue related to this PR. Maybe it isn't need? But I found that some tests failed if that line was omitted.